### PR TITLE
Feature/map reset

### PR DESCRIPTION
**Pull request by PawanKaur**
For issue #98 and #92 

1) Reset the values to Germany: A reset icon is added at the search bar
2) Reset zoom level. A home icon is added to the map that resets the zoom level to the default. It also changes the value to Germany wide. Till the time highlight district is not activated on the map, this functionality is needed else, its difficult to track the selected district on the map view.

https://gist.github.com/NXXR/5f71f73b3259769a336728629017d600

### CODE REVIEW

File: frontend/src/components/ScenarioCard.tsx [hunk 1]
```diff
@@ -7,6 +7,9 @@ import {Dictionary} from '../util/util';
 import {useTranslation} from 'react-i18next';
 import {NumberFormatter} from '../util/hooks';
 import {CheckBoxOutlineBlank, CheckBox} from '@mui/icons-material';
+import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import ArrowRightIcon from '@mui/icons-material/ArrowRight';
```
Comment (NXXR) [inline frontend/src/components/ScenarioCard.tsx:12]: These are changes from PR #149 right?
```suggestion
```

File: frontend/src/components/ScenarioCard.tsx [hunk 2]
```diff
@@ -207,6 +210,7 @@ export default function ScenarioCard(props: ScenarioCardProps): JSX.Element {
             {compartments.map((compartment, i) => (
               // hide compartment if expandProperties false and index > 4
               // highlight compartment if selectedProperty === compartment
+
```
Comment (NXXR) [inline frontend/src/components/ScenarioCard.tsx:213]: These are changes from PR #149 right?
```suggestion
```

File: frontend/src/components/ScenarioCard.tsx [hunk 3]
```diff
@@ -239,6 +243,37 @@ export default function ScenarioCard(props: ScenarioCardProps): JSX.Element {
...
+                  // shows grey arrows (stagnation) if getCompartmentRate is between 0 and 3 %. If there is no RKI value and then it checks for getCompartmentValue.
+                  sx={{
+                    display:
+                      (parseFloat(getCompartmentRate(compartment)) >= 0 &&
+                        parseFloat(getCompartmentRate(compartment)) <= 3) ||
+                      parseFloat(getCompartmentValue(compartment)) === 0
+                        ? 'block'
+                        : 'none',
+                  }}
+                ></ArrowRightIcon>
```
Comment (NXXR) [inline frontend/src/components/ScenarioCard.tsx:276]: These are changes from PR #149 right?
```suggestion
```

File: frontend/src/components/Sidebar/DistrictMap.tsx [hunk 4]
```diff
@@ -111,7 +112,7 @@ export default function DistrictMap(): JSX.Element {
     const chart = root.container.children.push(
       am5map.MapChart.new(root, {
         projection: am5map.geoMercator(),
-        maxZoomLevel: 4,
+        maxZoomLevel: 5,
```
Comment (NXXR) [inline frontend/src/components/Sidebar/DistrictMap.tsx:115]: Is there a reason for the changed zoom level?

File: frontend/src/components/Sidebar/DistrictMap.tsx [hunk 5]
```diff
@@ -275,12 +276,28 @@ export default function DistrictMap(): JSX.Element {
               legendRef.current = legend;
             }}
             min={0}
-            max={legend.isNormalized ? aggregatedMax : legend.steps[legend.steps.length - 1].value}
+            // used math.round to convert the number in integers and used +1 to avoid having legend max value as 0
+            max={
+              legend.isNormalized
+                ? Math.round(aggregatedMax) + 1
+                : Math.round(legend.steps[legend.steps.length - 1].value) + 1
+            }
```
Comment (NXXR) [inline frontend/src/components/Sidebar/DistrictMap.tsx:284]: This is also part of a different PR right?
```suggestion
            max={legend.isNormalized ? aggregatedMax : legend.steps[legend.steps.length - 1].value}
```

File: frontend/src/components/Sidebar/DistrictMap.tsx [hunk 6]
```diff
@@ -275,12 +276,28 @@ export default function DistrictMap(): JSX.Element {
...
         </Grid>
         <Grid item container justifyContent='center' xs={1}>
+          <ZoomOutMapIcon
+            id='home'
+            color='primary'
+            fontSize='large'
+            // if the chart is zoomed then, it will bring the chart to its default state, plus change the values to Germany wide.
+
+            onClick={() => {
+              dispatch(selectDistrict({ags: '00000', name: t('germany'), type: ''}));
```
Comment (NXXR) [inline frontend/src/components/Sidebar/DistrictMap.tsx:297]: why are we chanign the selected district too on the zoom out button?
To me it seems more intuitive if the Zoom out button just zooms and pans the map back into its default position.

Or did we have another reason for this?
```suggestion
```

**Review by NXXR — CHANGES_REQUESTED** [review]
some changes from other PRs mixed in again.
Also, the zoom out button also selects germany again, I'm confused on what we decided on for the interaction.

**Review by NXXR — CHANGES_REQUESTED** [review]
The size of the gemany icon/button does not match the size of the other two buttons.
or is that just with my build?

**Comment by PawanKaur** [comment]
> The size of the gemany icon/button does not match the size of the other two buttons. or is that just with my build?

I am not sure how it looks at your end, but at my side, it does not look much bigger or different.

**Comment by PawanKaur** [comment]
> Also make sure to pull the newest changes of the develop branch and resolve any issues blocking the merge :)
Done

**Comment by NXXR** [comment]
> > The size of the gemany icon/button does not match the size of the other two buttons. or is that just with my build?
> 
> I am not sure how it looks at your end, but at my side, it does not look much bigger or different.

grafik

This is how it looks on my end.
I was talking with Jonas to check if it was a problem on my end.

I think we should try to get the button into the map controls container so it's closer and in the same style as the zoom buttons.

This way it doesn't look out of place between the zoom controls and the heat legend buttons.

If we can't get it into the zoom controls we should try to at least match the style.

**Review by NXXR — CHANGES_REQUESTED** [review]
see comment above.

**Pull request closed without merging**